### PR TITLE
Always use Python 3.11 in pixi

### PR DESCRIPTION
**Pull request by emilk**
### What
We should all use the same Pyhthon version for linting, otherwise we will get different results. For instance, I get 60 errors running `pixi run lint-py-mypy` on `main`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5365/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5365/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5365/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5365)
- [Docs preview](https://rerun.io/preview/77a3d91553a7b50eb14b9acb54852d04be4efa1c/docs) 
- [Examples preview](https://rerun.io/preview/77a3d91553a7b50eb14b9acb54852d04be4efa1c/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Review by abey79 — CHANGES_REQUESTED** [review]
Fairly strong reject on my part. As noted on slack, this can cause lots of issues. For example: this caps mypy to a version that supports 3.8, which might be old, and less stringent than more recent mypy version.

My general rule of thumb:
- lint/format/whatnot with the most recent supported version
- run tests on _all_ supported versions

**Comment by emilk** [comment]
Ok, I've changed it to always run Python 3.12 in pixi instead, and to run our `nox` "test on all python versions" on CI

**Comment by emilk** [comment]
This fixes `pixi run lint-py-mypy` for me, which used to return 60 errors before

File: .github/workflows/contrib_rerun_py.yml [hunk 1]
```diff
@@ -100,6 +100,11 @@ jobs:
         shell: bash
         run: cd rerun_py/tests && pytest -c ../pyproject.toml
 
+      # TODO(#5332): only run this nightly or on main
+      - name: Run Python tests on all Python versions
+        shell: bash
+        run: nox -s tests
```
Comment (abey79) [inline .github/workflows/contrib_rerun_py.yml:106]: Are you sure nox is installed by default in the runner? I dont think it's in the `requirements-*.txt` as this should not depend on a venv (nox actually creates venv for each subtasks, like pixi).

Comment (abey79) [reply .github/workflows/contrib_rerun_py.yml:106]: Also, yeah, this takes quite some time, so it might be really annoying to have on PRs

**Review by abey79 — APPROVED** [review]
Nice!

File: .github/workflows/contrib_rerun_py.yml (hunk 1 above)
Comment (emilk) [reply .github/workflows/contrib_rerun_py.yml:106]: this is the CI - of course I'm not sure

Comment (emilk) [reply .github/workflows/contrib_rerun_py.yml:106]: I put this in the completely wrong file too

Comment (emilk) [reply .github/workflows/contrib_rerun_py.yml]: I removed it again. We'll do it in https://github.com/rerun-io/rerun/issues/2445 instead, any year now

**Comment by emilk** [comment]
Trying to get `pixi run pip install -r examples/python/requirements.txt` to work…

**Comment by emilk** [comment]
Went down to 3.11 to get CI green, and now `pixi run lint-py-mypy` fails again

**Pull request merged**